### ISSUE TITLE
Allows an alter an event and allows to stop sending it as well

### ISSUE THREAD

**Issue by rthideaway** [Gizra/logs_http#18]
We would need a feature to extend an event array and also have an ability to completely prevent sending to the log service. Mostly because of environments which are not able to send the events.

At the moment I patched it with this [patch](https://gist.githubusercontent.com/rthideaway/67c922762b62d18edef7a8f0cca89c89/raw/4b6f40fb1917cc5588e8d3da068660d71b59e19d/gistfile1.txt). Allowing modules to alter an event and also send a flag to prevent sending completely.

**Comment by RoySegall** [comment]
You should do something like:
```
if (!empty($event['send']) && !$event['send']) {
  // Do something awesome.
}
```

Because there could be a case where the `send` key does not exist in the array.

**Comment by rthideaway** [comment]
@RoySegall first of all thanks for pointing that out! I have it wrong, true, but I think the following would be enough for this particular case:
```
if (empty($event['send'])) {
  continue;
}
```
Simply if $event['send'] is not set to something positive, then we skip sending event to the log service.

**Comment by RoySegall** [comment]
So, in that case, we need to make sure that all the events will have the `send` property set to `TRUE` because now, the item is not initialized anywhere and that mean that all the events will be skipped by default and we don't want that, right? :wink:

**Comment by rthideaway** [comment]
@RoySegall I hope the link to my patch isn't broken, but actually all the events already have `send` to `TRUE`:
```
+    $event['send'] = TRUE;
+    $module_handler = \Drupal::moduleHandler();
+    $module_handler->alter('logs_http_event', $event);
+
+    // If event does not want be sent, continue with the next one.
+    if (empty($event['send'])) {
+      continue;
+    }
```

**Comment by RoySegall** [comment]
I think that's OK but there are 2 missing:
1. We need a test to make sure the functionality will work
2. Make this as a PR so we could merge this properly.

Except that, @ordavidil Any thoughts on this one?

**Comment by rthideaway** [comment]
@RoySegall PR: https://github.com/Gizra/logs_http/pull/20
I'll try to cover this with a test once I have a bit of spare time. Possibly during weekend.

**Comment by RoySegall** [comment]
looks awesome. Let's wait for tests.

**Comment by fago** [comment]
I just wondered why this is not done by overriding the "enabled" flag in the config via settings.php? This should achieve the same?

**Comment by rthideaway** [comment]
@fago If you would have config split per environment then yes. But if you have one config for all environments and you want to disable sending messages only on develop, then you have to keep your logs enabled and just disable sending to server. That was idea behind that flag.

**Comment by rthideaway** [comment]
@fago my bad, so you mean to override config in settings.php with
```$config['logs_http.settings']['enabled'] = TRUE;```
That should work yes :D But I still think that flag may be useful in some cases when you want disable sending messages in some specific case. Like ignoring specific "Info" message.

**Comment by nikunjkotecha** [comment]
for a commerce based project it is really important to keep all the messages around checkout and even for other projects it is possible that we want to keep messages of the all the types only for particular scenario and not for each and every message (for instance, info messages when user login but no messages around successful cron jobs)

+1 to this issue and IMO even solution is generic enough and would allow custom modules to control what to log and what not to log.

**Comment by scottsawyer** [comment]
Yep, this would be awesome, then I could create a module that would allow filtering logged events based on type and / or severity.
